src/org/omegat/util/gui/DesktopWrapper.java

```diff
@@ -0,0 +1,88 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+ with fuzzy matching, translation memory, keyword search,
+ glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2021 Lev Abashkin
+ Home page: http://www.omegat.org/
+ Support center: https://omegat.org/support
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.util.gui;
+
+
+import org.omegat.util.Platform;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Wrapper class for java.awt.Desktop with `xdg-open` path on Linux platform.
+ */
+public class DesktopWrapper {
+
+    private final static Desktop awtDesktop = Desktop.getDesktop();
```

Comment:
I'm not so sure about caching this. Are we so sure that the `Desktop` object can't ever change, and that it never will in the future?

None of these methods should ever be called in a loop, so I would be fine calling `Desktop.getDesktop()` every time, until there is evidence that doing so is somehow causing a problem.

Reply:
I hesitated about this one, because it's unlikely to change during program run, but I agree that this is an over-optimization.